### PR TITLE
Ping Pong Tests using four tools

### PR DESCRIPTION
**Pull request by prudmiry**
Ping Pong Tests using four tools

ibv_ud_pingpong,ibv_uc_pingpong,ibv_rc_pingpong,ibv_srq_pingpong

Signed-off-by: prudhvi <mprudhvi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by prudmiry** [comment]
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/508215/job.txt)
log file added

File: io/net/infiniband/ib_pingpong.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        self.PORT = int(self.params.get("PORT_NUM", default="1"))
+        self.PEER_CA = self.params.get("PEERCA", default="mlx4_0")
+        self.PEER_GID = int(self.params.get("PEERGID", default="0"))
+        self.PEER_PORT = int(self.params.get("PEERPORT", default="1"))
+        self.to = self.params.get("timeout", default="120")
+
+    def pingpong_exec(self, arg1, arg2, arg3):
+        '''
+            ping pong exec function
+        '''
```
Comment (PraveenPenguin) [inline io/net/infiniband/ib_pingpong.py]: can you please reomve this white space 
can be like 
```
      '''
      ping pong exec function
       '''
```

File: io/net/infiniband/ib_pingpong.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        msg = "timeout %s cat /tmp/ib_log; rm -rf /tmp/ib_log" % self.to
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True) != 0:
+            self.fail("test failed")
+
+    def test_ib_pingpong(self):
+        '''
+           test options are mandatory
+           ext test options are depends upon user
+        '''
```
Comment (PraveenPenguin) [inline io/net/infiniband/ib_pingpong.py]: please here as well 
```
'''
{}
'''
```

File: io/net/infiniband/ib_pingpong.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            self.pingpong_exec(tool_name, val1, val2)
+        ext_test_op = self.params.get("ext_test_opt", default="").split(",")
+        if self.flag == "1":
+            for val in ext_test_op:
+                self.pingpong_exec(tool_name, val, "")
+        else:
+            self.log.info("Extended test option skipped")
+        '''
+            change MTU to 1500 for non-IB tests
+        '''
```
Comment (PraveenPenguin) [inline io/net/infiniband/ib_pingpong.py]: please here as well

File: io/net/infiniband/ib_pingpong.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        test = arg2
+        logs = "> /tmp/ib_log 2>&1 &"
+        if test == "basic":
+            test = ""
+        msg = "timeout %s %s -d %s -g %d -i %d %s %s %s" \
+            % (self.to, arg1, self.PEER_CA, self.PEER_GID, self.PEER_PORT,
+                test, arg3, logs)
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True) != 0:
```
Comment (PraveenPenguin) [inline io/net/infiniband/ib_pingpong.py]: ignore_status=True parameter in process.system() will be used if process.system() returns a non-zero value, the script will throw a error like 

```
  File "/usr/lib/python2.7/site-packages/avocado_framework-41.0-py2.7.egg/avocado/core/test.py", line 510, in _run_avocado
    raise test_exception
```
` if process.system(cmd, shell=True,ignore_status=True) != 0:`

File: io/net/infiniband/ib_pingpong.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            self.fail("ssh failed to remote machine")
+        time.sleep(2)
+        self.log.info("client data for %s(%s)" % (arg1, arg2))
+        self.log.info("%s -d %s -g %d %s -i %d %s %s"
+                      % (arg1, self.CA, self.GID, self.PEER_IP, self.PORT,
+                         test, arg3))
+        tmp = "timeout %s %s -d %s -g %d -i %d %s %s %s" \
+            % (self.to, arg1, self.CA, self.GID, self.PORT, self.PEER_IP,
+                test, arg3)
+        if process.system(tmp, shell=True) != 0:
```
Comment (PraveenPenguin) [inline io/net/infiniband/ib_pingpong.py]: please here as well

`if process.system(tmp, shell=True,ignore_status=True) != 0:`

File: io/net/infiniband/ib_pingpong.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            self.fail("test failed")
+        self.log.info("server data for %s(%s)" % (arg1, arg2))
+        self.log.info("%s -d %s -g %d -i %d %s %s"
+                      % (arg1, self.PEER_CA, self.PEER_GID, self.PEER_PORT,
+                         test, arg3))
+        msg = "timeout %s cat /tmp/ib_log; rm -rf /tmp/ib_log" % self.to
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True) != 0:
```
Comment (PraveenPenguin) [inline io/net/infiniband/ib_pingpong.py]: please here as well :)

**Comment by prudmiry** [comment]
closing current PR and raised new PR Ping Pong Tests #81

**Pull request closed without merging**